File: Chapter09/spring-boot-2-email-formatter/spring-boot-2-email-formatter-model/src/main/java/com/packtpub/spring/boot/email/formatter/model/repo/UserRepository.java
package com.packtpub.spring.boot.email.formatter.model.repo;

import com.packtpub.spring.boot.email.formatter.model.domain.User;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserRepository extends JpaRepository<User, Integer> {

    User findByEmail(String email);

    User findByUsername(String username);

}

File: Chapter05/spring-boot-2-movie-rating/src/main/kotlin/com/packtpub/springboot2movierating/repo/UserRepository.kt
/*
    Copyright (C) 2018  Shazin Sadakath
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
package com.packtpub.springboot2movierating.repo

import com.packtpub.springboot2movierating.model.User
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import reactor.core.publisher.Flux

interface UserRepository : ReactiveMongoRepository<User, Int> {
    fun findByUsername(username : String) : Flux<User>;
}
